### PR TITLE
Globe - line layer

### PR DESCRIPTION
**Pull request by kubapelc**
# Line layer adaptation for globe

Issue & discussion of globe:

https://github.com/maplibre/maplibre/issues/190
https://github.com/maplibre/maplibre/discussions/161

Additions and changes since [the previous PR](https://github.com/maplibre/maplibre-gl-js/pull/3882):

- Adapted line layer type for globe
- Added render tests for line layer on globe

[Main vector globe branch with all changes](https://github.com/windycom/maplibre-gl-js/tree/kubapelc/globe-vector)

A vector globe demo image

## Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
I've updated the globe branch from main and it seems it has conflicts with what you wrote, can you check if these can be resolved?

File: src/render/painter.ts [hunk 1]
```diff
@@ -448,16 +448,6 @@ export class Painter {
             this.renderToTexture.prepareForRender(this.style, this.transform.zoom);
             // this is disabled, because render-to-texture is rendering all layers from bottom to top.
             this.opaquePassCutoff = 0;
-
-            // update coords/depth-framebuffer on camera movement, or tile reloading
```
Comment (HarelM) [inline src/render/painter.ts:452]: I think this is an incorrect merge, is it?

Comment (HarelM) [reply src/render/painter.ts:452]: Looking at this again, I probably made the mistake when merging...

Comment (kubapelc) [reply src/render/painter.ts:452]: I think it's fine, the original painter.ts has those lines removed as well. The removed logic is handled in  `this.maybeDrawDepthAndCoords(false);` a few lines above.

**Pull request merged**